main.js

```diff
@@ -0,0 +1,142 @@
+import articleService from "./ArticleService.js";
+import productService from "./ProductService.js";
+
+class Product {
+  constructor(name, description, price, tags, images, favoriteCount = 0) {
+    this.name = name;
+    this.description = description;
+    this.price = price;
+    this.tags = tags;
+    this.images = images;
+    this._favoriteCount = favoriteCount;
+  }
+
+  set favoriteCount(Count) {
+    throw new Error("call favorite()");
+  }
+
+  get favoriteCount() {
+    return this._favoriteCount;
+  }
+
+  favorite() {
+    this._favoriteCount += 1;
+  }
+}
+
+class ElectronicProduct extends Product {
+  constructor(name, description, price, tags, favoriteCount = 0, manufactuer) {
+    super(name, description, price, tags, favoriteCount);
+    this.manufactuer = manufactuer;
+  }
+}
+
+class Article {
+  constructor(title, content, writer, likeCount = 0, createdAt) {
+    this.title = title;
+    this.content = content;
+    this.writer = writer;
+    this._likeCount = likeCount;
+    this.createdAt = new Date();
+  }
```

Comment:
```suggestion
class Article {
  constructor(title, content, writer, likeCount = 0) {
    this.title = title;
    this.content = content;
    this.writer = writer;
    this._likeCount = likeCount;
    this.createdAt = new Date();
  }
```
프로퍼티를 생서하고자 할 때 생성자의 매개 변수로 모든 변수를 넣을 필요가 없습니다. 특히나 생성자 함수에서 createdAt에 new Date()로 값을 할당하고 있으니 생성자로부터 받는 과정은 제거해도 될 것 같습니다.